### PR TITLE
style: modify pkg about database connection

### PR DESCRIPTION
**Pull request by whhe**
Move common process of database connection to 'pkg/util/database', three main changes:
- `pkg/oceanbase/connector/ob_connect_properties.go` -> `pkg/oceanbase/connector/datasource.go`
- `pkg/oceanbase/connector/ob_connector.go` -> `pkg/util/database/connector.go`
- `pkg/oceanbase/connector/ob_connector_manager.go` -> `pkg/util/database/pool.go`

### CODE REVIEW

File: pkg/util/database/connector.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import (
+	// register mysql driver
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/jmoiron/sqlx"
+)
+
+// Connector represents a connection pool.
+type Connector struct {
+	ds     DataSource
+	client *sqlx.DB
```
Comment (chris-sun-star) [inline pkg/util/database/connector.go:24]: The original implementation supports connection pool config of client, which means a connector is able to handle multiple db connections, this is useful when multiple goroutines use the same connector

File: pkg/util/database/connector.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/*
```
Comment (chris-sun-star) [inline pkg/util/database/connector.go:1]: database can be a package under pkg, no need to put it under util

Comment (whhe) [reply pkg/util/database/connector.go]: Done.

**Review by chris-sun-star — APPROVED** [review]
LGTM

**Pull request merged**